lib/widgets/adaptive_layout.dart

```diff
@@ -0,0 +1,38 @@
+import 'package:flutter/material.dart';
+
+/// AdaptiveLayout for adaptive ui.
+class AdaptiveLayout extends StatelessWidget {
+  const AdaptiveLayout({
+    Key? key,
+    required this.mobile,
+    required this.tablet,
+    required this.desktop,
+  }) : super(key: key);
+
+  final Widget mobile;
+  final Widget tablet;
+  final Widget desktop;
```

Comment:
Replace widgets with WidgetBuilder and you'll avoid holding widgets:
- buildMobile
- buildTablet
- buildDesktop